### DISCUSSION TITLE
Linux | Change "GrandOrgueConfig" file location.

### DISCUSSION THREAD

**Discussion by andrew-elkins** [GrandOrgue/grandorgue#1876] [Developers]
Hello,

I'm on Ubuntu and using the appimage version. 

It seems that under Settings > Paths most file paths can be directed elsewhere. 

However, I can't find a way to move ~/GrandOrgueConfig and ~/GrandOrgueConfig.last which are both under the home directory. 

Is there a way to move these files to another directory like:
`~/Applications/GrandOrgue/`

Thank you.

**Comment by kerkovits** [comment]
As far as I know, it is not configurable. In my opinion, it really should be in `~/.config/`.

**Comment by oleg68** [comment]
If we made the `GrandOrgueConfig` location configurable then where to store this option?

**Reply by andrew-elkins** [reply]
I like @kerkovits's answer `~/.config/ ` makes the most sense to me.  No other application on my system puts its config files in the home directory. So it feels out of place see the picture below:

![image](https://github.com/GrandOrgue/grandorgue/assets/90867984/d20466c9-2aae-486e-8954-3902874ff320)

**Reply by tkitynski** [reply]
XDG convention solves it nicely. If no XDG_CONFIG_HOME envar is available, the default is ${HOME}/.config. In there each application should create its folder, e.g. `GrandOrgue` (no dot in the front) and store all config files in there. For backward compatibility it should lookup `${HOME}/GrandOrgueConfig`, but only for reading in case of absence of before mentioned config dir, and possibly move the old config to the new directory.

**Comment by jaydenlo08** [comment]
Some file managers like Nautilus allow you to hide specific files using a ".hidden" file: create a new file at `~/.hidden`, append "GrandOrgueConfig"  and "GrandOrgueConfig.last" on each line, and they should act as hidden files. I guess this isn't the ideal solution, but it does the job of hiding the config files.

**Comment by larspalo** [comment]
Since 1f37e8c088108fe6dc14d82a288ceeb6759ff8cc there's a possibility for the user to specify where the config file would be placed by using -c or --config together with a path/file option (flag argument) when starting GO. This is of course a command line option that needs to be added to the starting command in order to override the default config file.
